Bottom: b88d986fce8f7d2a3fec89f874e2ed800125b46d
Top:    f1e2de638128e8eb84bb6a46bdf9f7e05fd3b58f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-24 18:53:45 +0000

Refresh of fix-win32-use-of-glutess-funct

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 494f1a7..d9e4e42 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -573,21 +573,17 @@ static GLenum tessVertexType;
 static int stashed_vertices;
 static int triangle_comp_idx;
 
+#ifndef CALLBACK
+#define CALLBACK
+#endif
 
-static void
+static void CALLBACK
 myError (GLenum errno)
 {
   printf ("gluTess error: %s\n", gluErrorString (errno));
 }
 
-static void
-myFreeCombined ()
-{
-  while (combined_num_to_free)
-    free (combined_to_free [-- combined_num_to_free]);
-}
-
-static void
+static void CALLBACK
 myCombine ( GLdouble coords[3], void *vertex_data[4], GLfloat weight[4], void **dataOut )
 {
 #define MAX_COMBINED_VERTICES 2500
@@ -618,7 +614,7 @@ myCombine ( GLdouble coords[3], void *vertex_data[4], GLfloat weight[4], void **
   *dataOut = new_vertex;
 }
 
-static void
+static void CALLBACK
 myBegin (GLenum type)
 {
   tessVertexType = type;
@@ -626,7 +622,7 @@ myBegin (GLenum type)
   triangle_comp_idx = 0;
 }
 
-static void
+static void CALLBACK
 myVertex (GLdouble *vertex_data)
 {
   static GLfloat triangle_vertices [2 * 3];
@@ -679,6 +675,13 @@ myVertex (GLdouble *vertex_data)
     printf ("Vertex received with unknown type\n");
 }
 
+static void
+myFreeCombined ()
+{
+  while (combined_num_to_free)
+    free (combined_to_free [-- combined_num_to_free]);
+}
+
 void
 hidgl_fill_polygon (int n_coords, Coord *x, Coord *y)
 {
